ios/FluentUI/List/ListItem.swift

```diff
@@ -0,0 +1,322 @@
+
+/// View that represents an item in a List.
+public struct ListItem<LeadingContent: View,
+                       TrailingContent: View,
+                       Title: StringProtocol,
+                       Subtitle: StringProtocol,
+                       Footer: StringProtocol>: View {
+
+	// MARK: Initializer
+
+	/// Creates a ListItem view
+	/// - Parameters:
+	///   - title: Text that appears as the first line of text
+	///   - subtitle: Text that appears as the second line of text
+	///   - footer: Text that appears as the third line of text
+	///   - leadingContent: The content that appears on the leading edge of the view
+	///   - trailingContent: The content that appears on the trailing edge of the view, next to the accessory type if provided
+    public init(title: Title,
+                subtitle: Subtitle = String(),
+                footer: Footer = String(),
+                @ViewBuilder leadingContent: @escaping () -> LeadingContent,
+                @ViewBuilder trailingContent: @escaping () -> TrailingContent) {
+        self.title = title
+        self.subtitle = subtitle
+        self.footer = footer
+        self.leadingContent = leadingContent
+        self.trailingContent = trailingContent
+        let layoutType = ListItem.layoutType(subtitle: subtitle, footer: footer)
+        self.tokenSet = ListItemTokenSet(customViewSize: { layoutType.leadingContentSize })
+    }
+
+    public var body: some View {
+        tokenSet.update(fluentTheme)
+
+        @ViewBuilder
+        var labelStack: some View {
+            let titleView = Text(title)
+                                .foregroundColor(Color(uiColor: tokenSet[.titleColor].uiColor))
+                                .font(Font(tokenSet[.titleFont].uiFont))
+                                .frame(minHeight: ListItemTokenSet.titleHeight)
+                                .lineLimit(titleLineLimit)
+                                .truncationMode(titleTruncationMode)
+                                .accessibilityIdentifier(AccessibilityIdentifiers.title)
+
+            switch layoutType {
+            case .oneLine:
+                titleView
+            case .twoLines, .threeLines:
+                let subtitleView = Text(subtitle)
+                    .foregroundColor(Color(uiColor: tokenSet[.subtitleColor].uiColor))
+                    .lineLimit(subtitleLineLimit)
+                    .truncationMode(subtitleTruncationMode)
+                    .accessibilityIdentifier(AccessibilityIdentifiers.subtitle)
+                VStack(alignment: .leading, spacing: ListItemTokenSet.labelVerticalSpacing) {
+                    titleView
+                    if layoutType == .twoLines {
+                        subtitleView
+                            .font(Font(tokenSet[.subtitleTwoLinesFont].uiFont))
+                            .frame(minHeight: ListItemTokenSet.subtitleTwoLineHeight)
+                    } else if layoutType == .threeLines {
```

Comment:
nit: Is the `if layoutType == .threeLines` needed? Since we're in `case .twoLines, .threeLines`, if it isn't `.twoLines`, doesn't it have to be `.threeLines`?

Reply:
addressed in #1876